src/meepgeom.cpp

```diff
@@ -1145,6 +1188,62 @@ void geom_epsilon::eff_chi1inv_matrix(meep::component c, symmetric_matrix *chi1i
 static int eps_ever_negative = 0;
 static meep::field_type func_ft = meep::E_stuff;
 
+struct matgrid_volavg {
+  meep::ndim dim;        // dimensionality of voxel
+  double rad;            // (spherical) voxel radius
+  double uval;           // bilinearly-interpolated weight at voxel center
+  double ugrad_abs;      // magnitude of gradient of uval
+  double beta;           // thresholding bias
+  double eta;            // thresholding erosion/dilation
+  double eps1;           // trace of epsilon tensor from medium 1
+  double eps2;           // trace of epsilon tensor from medium 2
+};
+
+static void get_uproj_w(const void *mgva_, double x0, double &uproj_, double &w_) {
+  matgrid_volavg *mgva = (matgrid_volavg *)mgva_;
```

Comment:
Since you also do this cast in the functions below, you should probably just pass `matgrid_volavg *mgva` directly to `get_uproj_w`.

Reply:
Fixed.